From 1b1ce981d485cdf88d19e51707fd1a95d19d075f Mon Sep 17 00:00:00 2001
From: Andy Shaw <andy.shaw@qt.io>
Date: Mon, 2 Nov 2020 10:05:07 +0100
Subject: [PATCH 058/171] Android: Don't use putIfAbsent as that is not
 available in older APIs

Fixes: QTBUG-88076
Change-Id: I1ab12c574be036babfd0e4aacb44d1f75c66f4ad
Reviewed-by: Assam Boudjelthia <assam.boudjelthia@qt.io>
(cherry picked from commit 2b4a581f34854d9c0bb8d53626d929abd0661bd8)
---
 src/android/jar/src/org/qtproject/qt5/android/QtNative.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/android/jar/src/org/qtproject/qt5/android/QtNative.java b/src/android/jar/src/org/qtproject/qt5/android/QtNative.java
index 563fc7bce6..13711f9d24 100644
--- a/src/android/jar/src/org/qtproject/qt5/android/QtNative.java
+++ b/src/android/jar/src/org/qtproject/qt5/android/QtNative.java
@@ -270,8 +270,8 @@ public class QtNative
         if (uri == null) {
             Log.e(QtTAG, "getSize(): No permissions to open Uri");
             return size;
-        } else {
-            m_cachedUris.putIfAbsent(contentUrl, uri);
+        } else if (!m_cachedUris.containsKey(contentUrl)) {
+            m_cachedUris.put(contentUrl, uri);
         }
 
         try {
-- 
2.31.1

